VideoLocker/src/main/java/org/edx/mobile/model/course/CourseStructureV1Model.java

```diff
@@ -17,21 +17,20 @@
     @SerializedName("root")
     public String root;
 
-    public  BlockModel getBlockById(String id){
+    public BlockModel getBlockById(String id) {
         return blockData.get(id);
     }
 
     /**
      * if parent is null, returns empty list.
      */
-    public List<BlockModel> getDescendants(BlockModel parent){
+    public List<BlockModel> getDescendants(BlockModel parent) {
         List<BlockModel> descendants = new ArrayList<>();
-        if (parent == null || !parent.isContainer() )
+        if (parent == null || !parent.isContainer() || null == parent.descendants)
```

Comment:
Maybe we could log a warning in this case, as this is an inconsistent state. Maybe do the same for the opposite case (non-container type and non-empty descendants) inside the `isContainer()` method.